api/app/GraphQL/Mutations/ArchivePool.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\Pool;
+use Database\Helpers\ApiEnums;
+use Error;
+use Illuminate\Support\Carbon;
+
+final class ArchivePool
+{
+    /**
+     * Closes the pool by setting the archived_at to now().
+     * @param  null  $_
+     * @param  array{}  $args
+     */
+    public function __invoke($_, array $args)
+    {
+        $pool = Pool::find($args['id']);
+        if ($pool->getStatusAttribute() !== ApiEnums::POOL_IS_CLOSED) {
+            throw new Error("You cannot archive a pool unless it is in the closed status.");
```

Comment:
Does anyone know how get an error to show up in the GraphQL response?  I think this will show up as a generic HTTP 500 error.  Not a huge deal since the UI will prevent it from happening but it seems like there should be a better way.

Reply:
Do you mean something like below?

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/3ac6c34b-e86b-4d5d-b6a9-c143ad572926)

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/c354383a-9feb-45c9-bd04-7f40edad5d68)

Reply:
Yes, that is exactly what I needed!  Thanks.
Ref: [validation errors](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7399/commits/39e7c859e160799abe3bac0816476670dbc6f6a5)